lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -401,6 +401,7 @@ def test__has_access_descriptor_beta_user(self):
     @ddt.data(None, YESTERDAY, TOMORROW)
     @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
     @patch('courseware.access_utils.get_current_request_hostname', Mock(return_value='preview.localhost'))
+    @patch('courseware.access_utils.get_current_request_port', Mock(return_value='8003'))
     def test__has_access_descriptor_in_preview_mode(self, start):
```

Comment:
Add a test of the negative case: If get_current_request_port returns, say `8000`, the user should not have access.